compiler/infra/OMRCfg.hpp

```diff
@@ -52,6 +52,7 @@ class TR_BlockCloner;
 class TR_BlockFrequencyInfo;
 class TR_ExternalProfiler;
 namespace TR { class Block; }
+namespace OMR { class Block; }
```

Comment:
Is this forward declaration really needed? It seems... out of place.

Reply:
Hi, the forward declaration is indeed needed because otherwise the `friend` declaration will throw an error. I noticed that the only place where `OMR::Block` is forward declared is in [`./compiler/il/OMRBlock.hpp:30`](https://github.com/eclipse/omr/blob/979f5de3c93dbf8b98ab9cd24442773883ee9437/compiler/il/OMRBlock.hpp#L28) where it is surrounded by some ifdefs. Should I place this forward declaration surrounded by [these ifdefs](https://github.com/eclipse/omr/blob/979f5de3c93dbf8b98ab9cd24442773883ee9437/compiler/infra/OMRCfg.hpp#L30) macros as well?

Reply:
Oh, that's right... because it's in a different namespace.

In that case, no. The extra ifdefs are not needed. However, I would prefer if this forward declaration was after all the `namespace TR` ones because it's kind of "special." A comment explaining why a forward declaration of something in the `OMR` namespace is needed would also be appropriate.